From 71f5d514ffe5ea2d4a4fe3c9c314043575f35f29 Mon Sep 17 00:00:00 2001
From: Andrey Vagin <avagin@openvz.org>
Date: Tue, 2 Jul 2013 19:14:01 +0400
Subject: [PATCH] posix-times: don't fail if /proc/PID/times is absent

/proc/PID/times was added in 3.10, but FC19 relesead with 3.9.

Usually we support only last version of kernel in the project git, so
we can commit it as an additional patch for the fedora package.

Cc: Adrian Reber <adrian@lisas.de>
Signed-off-by: Andrey Vagin <avagin@openvz.org>
---
 proc_parse.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/proc_parse.c b/proc_parse.c
index 28e2983..57311ec 100644
--- a/proc_parse.c
+++ b/proc_parse.c
@@ -1189,6 +1189,10 @@ int parse_posix_timers(pid_t pid, struct proc_posix_timers_stat *args)
 
 	file = fopen_proc(pid, "timers");
 	if (file == NULL) {
+		if (errno == ENOENT) {
+			pr_warn("Dump of posix timers isn't supported by this kernel\n");
+			return 0;
+		}
 		pr_perror("Can't open posix timers file!");
 		return -1;
 	}
-- 
1.8.3.1

